### PR TITLE
Type completeness foreign repo

### LINKED ISSUE #3541: Make Type Completeness workflow work for PRs from external repos

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Closes #3541

### CODE REVIEW

File: .github/workflows/type_completeness.yml [hunk 1]
```diff
@@ -23,17 +23,18 @@ jobs:
...
+        # Must run before base completeness, as base completeness will checkout the base branch
+        # And we can't go back to the PR branch after that in case the PR is coming from a fork
         run: |
-          git checkout ${{ github.head_ref }}
           pip install . -U
           pyright --verifytypes telegram --ignoreexternal --outputjson > pr.json || true
           pyright --verifytypes telegram --ignoreexternal > pr.readable || true
+      - name: Get Base Completeness
+        run: |
+          git checkout ${{ github.base_ref }}
```
Comment (harshil21) [inline .github/workflows/type_completeness.yml:35]: what is github.base_ref pointing to? Where can I check its value?

**Review by harshil21 — COMMENTED** [review]
I don't know what the exact error message was btw

File: .github/workflows/type_completeness.yml (hunk 1 above)
Comment (Bibo-Joshi) [reply .github/workflows/type_completeness.yml:35]: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

**Comment by Bibo-Joshi** [comment]
> I don't know what the exact error message was btw

The issue can be seen e.g. in #3538

**Review by harshil21 — APPROVED** [review]
Ok looks good to me

**Pull request merged**